### DISCUSSION TITLE
Help with the Cake.AzCli - create a PR

### DISCUSSION THREAD

**Discussion by nick5454** [cake-build/cake#3418] [Extension Q&A]
I am trying to see how to use the [AzCli](https://github.com/lprichar/Cake.AzureCli) to create a PR with this [here](https://docs.microsoft.com/en-us/cli/azure/repos/pr?view=azure-cli-latest#az_repos_pr_create)

I assume this is correct `Az().Repos.Pr.Create(`

But, i'm confused on passing the args. Tested this locally and I know i need this command:

`az repos pr create --auto-complete false --description "merge sync" --org "https://dev.azure.com/orgname" --project "project" -r "repo" -s "dev_awaiting_merge" -t "develop" --title "Automated Merge"`

What type of list setting should I use or just pass arguments like AzLogin

`// "'az login' is accessed via Az().Login()
dynamic result = Az().Login(new LoginSettings {
   Username = username,
   // all commands can be customized if necessary with a ProcessArgumentBuilder
   Arguments = new ProcessArgumentBuilder()
      // anything appended with .AppendSecret() will be rendered as [REDACTED] if cake is run with `-verbosity=diagnostic`
      .Append("--password").AppendSecret(password)
});`

**Comment by nils-a** [comment]
As far as I can see Cake.AzureCli does not have `Az().Repos`. Maybe @lprichar can shed some light.

**Reply by lprichar** [reply]
Yes, sorry, I'm working on updating to the latest version of Azure CLI, but got distracted with client work.  I'll get back to this, this week and see if I can't get an updated version for you.

**Comment by pascalberger** [comment]
For working with Azure Repos there's the [Cake.AzureDevOps addin](https://cakebuild.net/extensions/cake-azuredevops/) which provides a [AzureDevOpsCreatePullRequest](https://cakebuild.net/api/Cake.AzureDevOps/AzureDevOpsAliases/4AE86944) alias
